### ISSUE TITLE
Crash with a scratch file with some ability stuff

### ISSUE THREAD

**Issue by alvaroc1** [unisonweb/unison#2495]
got a crash when saving a scratch file:
```
structural type Middleware2 e = Middleware2 (Text ->{e} Text)

doSomething : Text ->{IO} Text
doSomething x = x

> Middleware2 doSomething
```

```
  I found and typechecked these definitions in ~/workspace-ac/unison-scratch-v2/scratch.u. If you
  do an `add` or `update`, here's how your codebase would change:
  
    ⍟ These new definitions are ok to `add`:
    
      structural type Middleware2 e
      doSomething : Text ->{IO} Text
  
  Now evaluating any watch expressions (lines starting with `>`)... Ctrl+C cancels.

unison: DatabaseIntegrityError (NoObjectForPrimaryHashId (HashId 26760)) in getWatch
CallStack (from HasCallStack):
  error, called at src/Unison/Codebase/SqliteCodebase.hs:296:88 in unison-parser-typechecker-0.0.0-67l3hAIUJMyBSSQOYSrf3j:Unison.Codebase.SqliteCodebase
  getCycleLen, called at src/Unison/Codebase/SqliteCodebase.hs:572:34 in unison-parser-typechecker-0.0.0-67l3hAIUJMyBSSQOYSrf3j:Unison.Codebase.SqliteCodebase
```

**Comment by pchiusano** [comment]
This is a known issue that should get fixed soon when watch expressions evaluate to a constructor that isn't in the codebase. In the meantime, it should work after adding `Middleware2` to your codebase.

**Comment by ChrisPenner** [comment]
Confirmed that this one is fixed by #2471  ✅

**Comment by ChrisPenner** [comment]
Done in #2627 🎉
